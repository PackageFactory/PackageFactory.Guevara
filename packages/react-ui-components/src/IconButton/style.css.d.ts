export interface IStyleCss {
  'iconButton': string;
  'iconButton--disabled': string;
  'size-small': string;
}

export const locals: IStyleCss;
