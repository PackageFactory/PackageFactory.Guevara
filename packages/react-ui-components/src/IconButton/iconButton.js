import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import mergeClassNames from 'classnames';

class IconButton extends PureComponent {
    static propTypes = {
        /**
         * The icon key which gets passed to the Icon Component.
         */
        icon: PropTypes.string.isRequired,

        /**
         * An optional `className` to attach to the wrapper.
         */
        className: PropTypes.string,

        /**
         * Defines the size of the icon button.
         */
        size: PropTypes.oneOf(['small', 'regular']),

        /**
        * An optional css theme to be injected.
        */
        theme: PropTypes.shape({/* eslint-disable quote-props */
            'iconButton': PropTypes.string
        }).isRequired, /* eslint-enable quote-props */

        /**
         * Static component dependencies which are injected from the outside (index.js)
         */
        IconComponent: PropTypes.any.isRequired,
        ButtonComponent: PropTypes.any.isRequired,

        /**
         * Optional disabled flag
         */
        disabled: PropTypes.bool
    };

    static defaultProps = {
        size: 'regular',
        style: 'transparent',
        hoverStyle: 'brand'
    };

    render() {
        const {
            IconComponent,
            ButtonComponent,
            className,
            theme,
            icon,
            size,
            disabled,
            ...rest
        } = this.props;
        const finalClassName = mergeClassNames({
            [className]: className && className.length,
            [theme.iconButton]: true,
            [theme[`size-${size}`]]: true,
            [theme['iconButton--disabled']]: disabled
        });

        return (
            <ButtonComponent {...rest} size={size} className={finalClassName}>
                <IconComponent icon={icon}/>
            </ButtonComponent>
        );
    }
}

export default IconButton;
