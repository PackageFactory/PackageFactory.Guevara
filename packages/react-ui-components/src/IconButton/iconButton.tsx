import {IconName} from '@fortawesome/fontawesome-svg-core';
import mergeClassNames from 'classnames';
import React, {PureComponent} from 'react';

import {PickDefaultProps} from '../../types';
import {ButtonHoverStyle, ButtonProps, ButtonSize, ButtonStyle} from '../Button/button';
import {IconProps} from '../Icon/icon';

interface IconButtonTheme {
    readonly 'iconButton': string;
    readonly [key: string]: string;
}

interface IconButtonProps {
    /**
     * The icon key which gets passed to the Icon Component.
     */
    readonly icon: string | IconName;

    /**
     * An optional `className` to attach to the wrapper.
     */
    readonly className?: string;

    /**
     * Defines the size of the icon button.
     */
    readonly size: ButtonSize;

    /**
     * An optional css theme to be injected.
     */
    readonly theme?: IconButtonTheme;

    // TODO: This feels strange. Can we use the actual class here?
    /**
     * Static component dependencies which are injected from the outside (index.js)
     */
    readonly IconComponent: React.ComponentClass<IconProps>;
    readonly ButtonComponent: React.ComponentClass<ButtonProps>;

    /**
     * Optional disabled flag
     */
    readonly disabled?: boolean;

    readonly style: ButtonStyle;

    readonly hoverStyle: ButtonHoverStyle;
}

type DefaultProps = PickDefaultProps<IconButtonProps, 'hoverStyle' | 'size' | 'style'>;

const defaultProps: DefaultProps = {
    hoverStyle: 'brand',
    size: 'regular',
    style: 'transparent',
};

class IconButton extends PureComponent<IconButtonProps> {
    public static readonly defaultProps = defaultProps;

    public render(): JSX.Element {
        const {
            IconComponent,
            ButtonComponent,
            className,
            theme,
            icon,
            size,
            disabled,
            ...rest
        } = this.props;
        const finalClassName = mergeClassNames(
            className,
            theme!.iconButton,
            theme![`size-${size}`],
            {
                [theme!['iconButton--disabled']]: disabled
            }
        );

        return (
            <ButtonComponent {...rest} size={size} className={finalClassName}>
                <IconComponent icon={icon}/>
            </ButtonComponent>
        );
    }
}

export default IconButton;
