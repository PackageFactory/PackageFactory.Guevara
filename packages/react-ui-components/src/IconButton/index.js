import {themr} from '@friendsofreactjs/react-css-themr';
import identifiers from './../identifiers';
import style from './style.css';
import IconButton from './iconButton.js';

const ThemedIconButton = themr(identifiers.iconButton, style)(IconButton);

//
// Dependency injection
//
import injectProps from './../_lib/injectProps.js';
import Icon from './../Icon/index';
import Button from './../Button/index';

export default injectProps({
    IconComponent: Icon,
    ButtonComponent: Button
})(ThemedIconButton);
