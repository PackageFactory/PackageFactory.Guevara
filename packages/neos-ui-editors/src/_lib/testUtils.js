import {PureComponent, Children} from 'react';
import PropTypes from 'prop-types';
import {SynchronousRegistry, SynchronousMetaRegistry} from '@neos-project/neos-ui-extensibility/src/registry';

const globalRegistry = new SynchronousMetaRegistry();

// i18n Registry
class FakeI18NRegistry extends SynchronousRegistry {
    translate(key) {
        return key;
    }
}
globalRegistry.set('i18n', new FakeI18NRegistry());

// Data Source DataLoader
export const MockDataSourceDataLoader = {
    _currentPromise: null,
    _currentPromiseResolveFn: null,

    // API in tests
    reset() {
        this._currentPromise = null;
    },

    resolveCurrentPromise(result) {
        this._currentPromiseResolveFn(result);
        return this._currentPromise;
    },

    // API of DataLoader
    resolveValue() {
        if (!this._currentPromise) {
            this._currentPromise = new Promise(resolve => {
                this._currentPromiseResolveFn = resolve;
            });
        }
        return this._currentPromise;
    }
};
globalRegistry.set('dataLoaders', new SynchronousRegistry());
globalRegistry.get('dataLoaders').set('DataSources', MockDataSourceDataLoader);

// wrapper
export class WrapWithMockGlobalRegistry extends PureComponent {
    static propTypes = {
        children: PropTypes.element.isRequired
    };

    static childContextTypes = {
        globalRegistry: PropTypes.object.isRequired,
        configuration: PropTypes.object.isRequired
    };

    getChildContext() {
        const configuration = {};
        return {globalRegistry, configuration};
    }

    render() {
        return Children.only(this.props.children);
    }
}
