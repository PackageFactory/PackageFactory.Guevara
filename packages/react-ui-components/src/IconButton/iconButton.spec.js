import {createShallowRenderer, createStubComponent} from './../_lib/testUtils.js';
import IconButton from './iconButton.js';

const Icon = createStubComponent();
const Button = createStubComponent();
const defaultProps = {
    IconComponent: Icon,
    ButtonComponent: Button,
    icon: 'fooIconName',
    theme: {/* eslint-disable quote-props */
        'iconButton': 'iconButtonClassName'
    }/* eslint-enable quote-props */
};
const shallow = createShallowRenderer(IconButton, defaultProps);

test('should render the "iconButton" className prop of the theme.', () => {
    const wrapper = shallow();

    expect(wrapper.hasClass('iconButtonClassName')).toBeTruthy();
});
test('should render the "className" if provided.', () => {
    const wrapper = shallow({className: 'fooClassName'});

    expect(wrapper.hasClass('fooClassName')).toBeTruthy();
});
test('should render the "ButtonComponent" and propagate all props to it.', () => {
    const wrapper = shallow({className: 'fooClassName'});
    const btn = wrapper.find(Button);

    expect(btn.hasClass('fooClassName')).toBeTruthy();
});
test('should render the "IconComponent" and propagate the "icon" prop to it.', () => {
    const wrapper = shallow({className: 'fooClassName'});
    const icon = wrapper.find(Icon);

    expect(icon.prop('icon')).toBe('fooIconName');
});
