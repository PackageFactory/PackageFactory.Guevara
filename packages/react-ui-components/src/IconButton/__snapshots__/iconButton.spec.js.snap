// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton/> should render correctly. 1`] = `
<Component
  className="iconButtonClassName"
  hoverStyle="brand"
  size="regular"
  style="transparent"
>
  <Component
    icon="fooIconName"
  />
</Component>
`;
